### PR TITLE
feat(indexers): SubsPlease parse title

### PR DESCRIPTION
**Pull request by danwilldev**
This change aims to parse any multiple-word title to populate the title macro.

### CODE REVIEW

**Comment by danwilldev** [comment]
An earlier discussion made me realize that subsplease doesn't parse the title. This change would be useful for folder naming

**Comment by martylukyy** [comment]
Merging this would probably result in the same problems mentioned in https://github.com/autobrr/autobrr/pull/1173

File: internal/indexer/definitions/subsplease.yaml [hunk 1]
```diff
@@ -46,11 +46,13 @@ irc:
     type: single
     lines:
       - test: 
-          - "[Release] [SubsPlease] Title - 01 (480p) [636F636B].mkv (420.69MB) - https://nyaa.si/view/0000000 - https://nyaa.si/view/0000000/torrent"
-        pattern: '\[Release\] (.*(SubsPlease).*?)\.?(mkv)? \((\d+.?\d*[KMGTP]?B)\) - (.*) - (.*)'
+          - "[Release] [SubsPlease] Multiple Word Title - 01 (480p) [636F636B].mkv (420.69MB) - https://nyaa.si/view/0000000 - https://nyaa.si/view/0000000/torrent"
+          - "[Release] [SubsPlease] Multiple Word Title (01-12) (1080p) [Batch] (4.95GB) - https://nyaa.si/view/0000000 - https://nyaa.si/view/0000000/torrent"
+        pattern: '\[Release\] (.*(SubsPlease)] ([^-()]+) .*?\.?)(mkv)? \((\d+.?\d*[KMGTP]?B)\) - (.*) - (.*)'
```
Comment (martylukyy) [inline internal/indexer/definitions/subsplease.yaml:51]: Creating a character class which prohibits matching of hyphens or dashes breaks matching of all titles with said characters.

**Review by martylukyy — CHANGES_REQUESTED** [review]

File: internal/indexer/definitions/subsplease.yaml (hunk 1 above)
Comment (danwilldev) [reply internal/indexer/definitions/subsplease.yaml]: Used pattern matching instead of character class in the latest commit

**Review by martylukyy — APPROVED** [review]
Tested with approx. 4700 announces should be good.

**Review by zze0s — APPROVED** [review]

**Pull request merged**